### PR TITLE
Add support for blackouts with regex

### PR DESCRIPTION
**Pull request by headphonejames**
Blackout Regex alerta plugin. Allow regex blackouts to be applied to alerts. Original code from: https://github.com/mirceaulinic/alerta-blackout-regex. Updated and fixed for both postgres and mongodb databases.

### CODE REVIEW

**Comment by kannon92** [comment]
Hey @headphonejames, is there a reason why we can't just reuse https://github.com/mirceaulinic/alerta-blackout-regex from pypi?

File: plugins/blackout-regex/test_blackout_regex.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+plugins_mod = sys.modules["alerta.plugins"] = MagicMock()
+plugins_mod.PluginBase = MagicMock
+
+
+from blackout_regex import BlackoutRegex  # pylama: ignore=E402
+
+log = logging.getLogger(__name__)
+
+
+class TestEnhance(unittest.TestCase):
```
Comment (kannon92) [inline plugins/blackout-regex/test_blackout_regex.py:144]: These tests seem to be a lot of duplicated code.  

Could you use something like https://docs.pytest.org/en/6.2.x/parametrize.html?

**Review by kannon92 — APPROVED** [review]
I do wonder why we can't just use the plugin from the repo.  And or fix that there.  

But other than that LGTM

**Review by richscott — APPROVED** [review]

**Comment by headphonejames** [comment]
@kannon92 our build process creates a docker image with all the required binaries and plug-ins. We could update the docker-alerta build script to pull in from this additional repo, but it does make it a little messier. 

I'm happy to try that approach, first fixing the plugin code in the original repo, then pulling in it during docker-alerta docker build install-plugins.sh script (which would require updating).

**Comment by kannon92** [comment]
> @kannon92 our build process creates a docker image with all the required binaries and plug-ins. We could update the docker-alerta build script to pull in from this additional repo, but it does make it a little messier.
> 
> I'm happy to try that approach, first fixing the plugin code in the original repo, then pulling in it during docker-alerta docker build install-plugins.sh script (which would require updating).

As an OSS team I like this approach. But I can understand why we won’t do it. No big deal either way. You should at least let the owner know if bugs you discovered

**Comment by headphonejames** [comment]
I did submit https://github.com/mirceaulinic/alerta-blackout-regex/pull/21 to address the bugs I fixed

**Pull request merged**